tools/log_analyzer.py

```diff
@@ -184,26 +233,87 @@ def item_creator():
         location: calculate_stddev(pickup_count, locations[location])
         for location in locations.keys()
     }
+    
+    regions = dict()
+    region_totals = dict()
+
+    total_progression_item_count = 0
+    for location in locations:
+        region = location.split("/")[0]
+
+        if region not in regions.keys():
+            regions[region] = 0
+
+        if region not in region_totals.keys():
+            region_totals[region] = 0
+        region_totals[region] += 1
+
+        count = 0
+        for item in locations[location]:
+            count = count + locations[location][item]
+        total_progression_item_count += count
+        regions[region] += count
+
+    # probability that any given location in this region contains progression
+    regions_weighted = dict()
+    for region in regions:
+        regions_weighted[region] = ((float(regions[region])/float(seed_count)) / float(region_totals[region]))
+
+    items = sort_by_contents(items)
+    locations = sort_by_contents(locations)
+    item_hints = sort_by_contents(item_hints)
+    location_hints = sort_by_contents(location_hints)
+
+    location_progression_count = {
+        location: value
+        for location, value in sorted(progression_count_for_location.items(), key=lambda t: t[1], reverse=True)
+    }
+
+    location_progression_no_key_count = {
+        location: value
+        for location, value in sorted(progression_no_key_count_for_location.items(),
+                                        key=lambda t: t[1], reverse=True)
+    }
+
+    for location in locations:
+        if location not in location_progression_no_key_count.keys():
+            location_progression_no_key_count[location] = 0
+    
+        if location not in location_progression_count.keys():
+            location_progression_count[location] = 0
+        
+        for item in items:
+            if location not in items[item].keys():
+                items[item][location] = 0
+
+    if use_percentage:
+        for item in items:
+            for room in items[item]:
+                items[item][room] = (float(items[item][room])/float(seed_count))
```

Comment:
no need to cast everything to floats, in Python 3 `/` is float division (`//` is for integer division)